include/msgpack/v1/adaptor/cpp17/variant.hpp

```diff
@@ -0,0 +1,126 @@
+//
+// MessagePack for C++ static resolution routine
+//
+// Copyright (C) 2023 Uy Ha
+//
+//    Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//    http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef MSGPACK_V1_TYPE_VARIANT_HPP
+#define MSGPACK_V1_TYPE_VARIANT_HPP
+
+#if defined(MSGPACK_USE_STD_VARIANT_ADAPTOR)
+
+#include "msgpack/cpp_version.hpp"
+
+#if MSGPACK_CPP_VERSION >= 201703
+
+#include "msgpack/adaptor/adaptor_base.hpp"
+#include "msgpack/object.hpp"
+#include "msgpack/versioning.hpp"
+
+#include <variant>
+
+namespace msgpack {
+MSGPACK_API_VERSION_NAMESPACE(v1) {
+  namespace adaptor {
+  namespace detail {
+  template <typename Variant, typename T, typename... Ts, std::size_t current_index, std::size_t... indices>
+  Variant construct_variant(std::size_t index,
+                            msgpack::object &object,
+                            std::index_sequence<current_index, indices...>) {
+    if constexpr(sizeof...(Ts) == 0) {
+      return object.as<T>();
+    } else {
```

Comment:
```suggestion
    } 
    else {
```

Please use this style. Not 100% complied, but at least new code do that.